app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt

```diff
@@ -229,188 +234,232 @@ class HealthConnectSensorManager : SensorManager {
 
     private suspend fun updateTotalCaloriesBurnedSensor(context: Context) {
         val healthConnectClient = getOrCreateHealthConnectClient(context) ?: return
-        val totalCaloriesBurnedRequest = healthConnectClient.aggregate(buildAggregationRequest(TotalCaloriesBurnedRecord.ENERGY_TOTAL))
-        val energy = totalCaloriesBurnedRequest[TotalCaloriesBurnedRecord.ENERGY_TOTAL]?.inKilocalories ?: 0.0
-        onSensorUpdated(
-            context,
-            totalCaloriesBurned,
-            BigDecimal(energy).setScale(2, RoundingMode.HALF_EVEN),
-            totalCaloriesBurned.statelessIcon,
-            attributes = buildAggregationAttributes(totalCaloriesBurnedRequest)
-        )
+        try {
+            val totalCaloriesBurnedRequest = healthConnectClient.aggregate(buildAggregationRequest(TotalCaloriesBurnedRecord.ENERGY_TOTAL))
+            val energy = totalCaloriesBurnedRequest[TotalCaloriesBurnedRecord.ENERGY_TOTAL]?.inKilocalories ?: 0.0
+            onSensorUpdated(
+                context,
+                totalCaloriesBurned,
+                BigDecimal(energy).setScale(2, RoundingMode.HALF_EVEN),
+                totalCaloriesBurned.statelessIcon,
+                attributes = buildAggregationAttributes(totalCaloriesBurnedRequest)
+            )
+        } catch (e: Exception) {
+            Log.e(TAG, "Unable to get total calories burned", e)
+        }
     }
 
     private suspend fun updateWeightSensor(context: Context) {
         val healthConnectClient = getOrCreateHealthConnectClient(context) ?: return
-        val weightRequest = buildReadRecordsRequest(WeightRecord::class) as ReadRecordsRequest<WeightRecord>
-        val response = healthConnectClient.readRecords(weightRequest)
-        if (response.records.isEmpty()) {
-            return
-        }
-        onSensorUpdated(
-            context,
-            weight,
-            BigDecimal(response.records.last().weight.inGrams).setScale(2, RoundingMode.HALF_EVEN),
-            weight.statelessIcon,
-            attributes = mapOf(
-                "date" to response.records.last().time,
-                "source" to response.records.last().metadata.dataOrigin.packageName
+        try {
```

Comment:
That's a lot of try/catch if you remember that there are tons of additional sensors we could add. What about an extension function:
```kotlin
    private suspend fun <T : Record> HealthConnectClient.readRecordsOrNull(request: ReadRecordsRequest<T>): ReadRecordsResponse<T>? = try {
        readRecords(request)
    } catch (e: Exception) {
        Log.e(TAG, "Could not read records", e)
        null
    }
```
In sensor code then you only need to update 2 lines:
```kotlin
    val response = healthConnectClient.readRecordsOrNull(weightRequest)
    if (response == null || response.records.isEmpty()) {
        return
    }
```

Reply:
this wont work each record has unique data which is why we have all those unchecked casts